core/client/app/mixins/ed-editor-api.js

```diff
@@ -0,0 +1,101 @@
+import Ember from 'ember';
+
+var EditorAPI = Ember.Mixin.create({
+    getValue: function () {
+        return this.get('element').value;
+    },
+
+    getSelection: function () {
+        return this.$().getSelection();
+    },
+
+    getLineToCursor: function () {
```

Comment:
Some docs on these api methods would be cool, but not crucial - just took me a moment to think of what "getLineToCursor" does